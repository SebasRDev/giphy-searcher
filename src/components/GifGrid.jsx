import React, { memo} from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import GifGridItem from './GifGridItem'
import PropTypes from 'prop-types' 

const GifGrid = ({category}) => {

  const {data:images, loading} = useFetchGifs( category );
  return (
    <>
      <h3 className='category-title aninate__animated animate__flash'>{category}</h3>
      { loading && <p>Cargando...</p>}
      <div className='card-grid'>
        {images.map(img => {
          return(
            <GifGridItem 
              key={img.id}
              {...img}
            />
          )
        })}
      </div>
    </>
  )
}

export default GifGrid

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}