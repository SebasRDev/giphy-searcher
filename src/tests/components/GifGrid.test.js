import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('test over GifGrid component', ()=>{
  const category = 'one punch man';

  test('Should match with snapshot', ()=>{
    useFetchGifs.mockReturnValue({
      data:[],
      loading: true
    })
    const wrapper = shallow(<GifGrid category={category}/>)
    expect(wrapper).toMatchSnapshot();
  })
  
  test('should show items when the images are loaded with useFecth', () => {
    
    const gifs = [{
      id:'ABC',
      url: 'https://localhost/cualquier/test.jpg',
      title: 'test'
    },
    {
      id:'DEF',
      url: 'https://localhost/cualquier/test.jpg',
      title: 'test'
    }];

    useFetchGifs.mockReturnValue({
      data:gifs,
      loading: false
    })

    const wrapper = shallow(<GifGrid category={category}/>);
    expect( wrapper.find('p').exists() ).toBe(false);
    expect( wrapper.find('GifGridItem').length ).toBe( gifs.length )

  })


})