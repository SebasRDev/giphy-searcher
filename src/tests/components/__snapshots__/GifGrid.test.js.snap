// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test over GifGrid component Should match with snapshot 1`] = `
<Fragment>
  <h3
    className="category-title aninate__animated animate__flash"
  >
    one punch man
  </h3>
  <p>
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
